src/cdk-experimental/listbox/listbox.spec.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  ComponentFixture,
+  async,
+  TestBed,
+} from '@angular/core/testing';
+import {Component, DebugElement} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {
+  CdkOption,
+  CdkListboxModule
+} from './index';
+import {dispatchMouseEvent} from '@angular/cdk/testing/private';
+
+describe('CdkOption', () => {
+
+  describe('selection state change', () => {
+    let fixture: ComponentFixture<CdkListboxWithCdkOptions>;
+    let options: DebugElement[];
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkListboxModule],
+        declarations: [CdkListboxWithCdkOptions],
+      }).compileComponents();
+    }));
+
+    beforeEach(async(() => {
+      fixture = TestBed.createComponent(CdkListboxWithCdkOptions);
+      fixture.detectChanges();
+      options = fixture.debugElement.queryAll(By.directive(CdkOption));
+    }));
+
+    it('should generate a unique optionId for each option', () => {
+      for (const option of options) {
+        expect(option.injector.get<CdkOption>(CdkOption).getOptionId()).toMatch(/cdk-option-\d+/);
+      }
+    });
+
+    it('should have set the selected input of the options to null by default', () => {
+      for (const option of options) {
+        expect(option.injector.get<CdkOption>(CdkOption).selected).toBeFalse();
+      }
+    });
+
+    it('should update aria-selected when selected is changed programmatically', () => {
+      expect(options[1].nativeElement.getAttribute('aria-selected')).toBeNull();
+      options[1].injector.get<CdkOption>(CdkOption).selected = true;
+      fixture.detectChanges();
+
+      expect(options[1].nativeElement.getAttribute('aria-selected')).toBe('true');
+    });
+
+    it('should update selected option on click event', () => {
```

Comment:
Should this test assert anything on the listbox instead of just the options?

Reply:
Yeah now that clicks trigger the event emitter on listbox, I'll test that the emit function was called.